src/modules/string/index.ts

```diff
@@ -0,0 +1,402 @@
+      );
+    }
+
+    return Array.from({ length: count }, () =>
+      this.faker.helpers.arrayElement(charsArray)
+    ).join('');
+  }
+
+  /**
+   * Generates a given length string of digits.
+   *
+   * @param options Either the number of characters or the options to use. Defaults to `{ length: 1, allowLeadingZeros = false, bannedDigits = [] }`.
+   * @param options.length The number of digits to generate. Defaults to `1`.
+   * @param options.allowLeadingZeros If true, leading zeros will be allowed. Defaults to `false`.
+   * @param options.bannedDigits An array of digits which should be banned in the generated string. Defaults to `[]`.
+   *
+   * @example
+   * faker.string.numeric() // '2'
+   * faker.string.numeric(5) // '31507'
+   * faker.string.numeric(42) // '56434563150765416546479875435481513188548'
+   * faker.string.numeric({ allowLeadingZeros: true, length: 42 }) // '00564846278453876543517840713421451546115'
+   * faker.string.numeric({ bannedDigits: ['0'], length: 6 }) // '943228'
+   *
+   * @since 8.0.0
+   */
+  numeric(
+    options:
+      | number
+      | {
+          length?: number;
+          allowLeadingZeros?: boolean;
+          bannedDigits?: readonly LiteralUnion<NumericChar>[] | string;
+        } = {}
+  ): string {
+    if (typeof options === 'number') {
+      options = {
+        length: options,
+      };
+    }
+
+    const { allowLeadingZeros = false, length = 1 } = options;
+    if (length <= 0) {
+      return '';
+    }
+
+    let { bannedDigits = [] } = options;
+
+    if (typeof bannedDigits === 'string') {
+      bannedDigits = bannedDigits.split('');
+    }
+
+    const allowedDigits = DIGIT_CHARS.filter(
+      (digit) => !bannedDigits.includes(digit)
+    );
+
+    if (
+      allowedDigits.length === 0 ||
+      (allowedDigits.length === 1 &&
+        !allowLeadingZeros &&
+        allowedDigits[0] === '0')
+    ) {
+      throw new FakerError(
+        'Unable to generate numeric string, because all possible digits are banned.'
+      );
+    }
+
+    let result = '';
+
+    if (!allowLeadingZeros && !bannedDigits.includes('0')) {
+      result += this.faker.helpers.arrayElement(
+        allowedDigits.filter((digit) => digit !== '0')
+      );
+    }
+
+    while (result.length < length) {
+      result += this.faker.helpers.arrayElement(allowedDigits);
+    }
+
+    return result;
+  }
+
+  /**
+   * Returns a UUID v4 ([Universally Unique Identifier](https://en.wikipedia.org/wiki/Universally_unique_identifier)).
+   *
+   * @example
+   * faker.string.uuid() // '4136cd0b-d90b-4af7-b485-5d1ded8db252'
+   *
+   * @since 8.0.0
+   */
+  uuid(): string {
+    const RFC4122_TEMPLATE = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx';
+    const replacePlaceholders = (placeholder: string) => {
+      const random = this.faker.datatype.number({ min: 0, max: 15 });
+      const value = placeholder === 'x' ? random : (random & 0x3) | 0x8;
```

Comment:
```suggestion
      const value = placeholder === 'x'
          ? this.faker.datatype.number({ min: 0, max: 15 }) 
          : this.faker.datatype.number({ min: 8, max: 11 });
```

Or however this should be formatted.

Reply:
Why is this implemented like this anyway? Can we just do 2 `replace` cycles on the template (one for 'x' and one for 'y')?

Reply:
It might be a neglible performance thing.
For me, it isn't that important.
What do the others think is best?

Reply:
Uh this looks heavy... 😖
Is it okay if I don't touch this for now in this PR?
Should we add a TODO, create an issue or just open another PR after this got merged?

Reply:
Extracted to https://github.com/faker-js/faker/issues/1466